server/src/main/java/pl/edu/agh/RaftServer.java

```diff
@@ -49,23 +52,28 @@
     private static final ScheduledExecutorService TIMEOUT_EXECUTOR = Executors.newSingleThreadScheduledExecutor();
 
     private final SocketAddress localAddress;
-    private final TcpServer<ByteBuf, ByteBuf> tcpServer;
-    private final Map<SocketAddress, Connection<ByteBuf, ByteBuf>> serverConnections;
+    private Map<SocketAddress, Connection<ByteBuf, ByteBuf>> serverConnections = Maps.newHashMap();
+    private Connection<ByteBuf, ByteBuf> clientConnection;
 
     private State state = State.FOLLOWER;
     private int currentTerm = 0;
     private AtomicInteger votesCount = new AtomicInteger(0);
     private SocketAddress votedFor;
     private ScheduledFuture timeout;
 
+    private volatile Map<String, Integer> keyValueStore = Maps.newHashMap();
```

Comment:
Sorry for the volatile keyword ;) I tend to use it whenever I deal with situations when we have 'multiple 'clients' wanting to change sth in variable (even if it's done by one server xD). Will change